nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/auth/ColdStorageS3TestHelper.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Abdoul BA<aba@nue.com>
+ */
+
+package org.nuxeo.coldstorage.auth;
+
+import java.util.Objects;
+
+import com.amazonaws.services.s3.model.CopyObjectRequest;
+import com.amazonaws.services.s3.model.StorageClass;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.coldstorage.helpers.ColdStorageHelper;
+import org.nuxeo.coldstorage.operations.AbstractTestColdStorageOperation;
+import org.nuxeo.ecm.blob.s3.S3BlobProvider;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.blob.BlobManager;
+import org.nuxeo.ecm.core.blob.ManagedBlob;
+import org.nuxeo.runtime.api.Framework;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ObjectListing;
+
+public class ColdStorageS3TestHelper {
+
+    protected static final Logger log = LogManager.getLogger(ColdStorageS3TestHelper.class);
+
+    protected static ColdStorageS3TestHelper INSTANCE;
+
+    protected static final String AWS_MAIN_BUCKET_NAME_ENV_VAR = "COLDSTORAGE_AWS_MAIN_BUCKET_NAME";
+
+    protected static final String AWS_GLACIER_BUCKET_NAME_ENV_VAR = "COLDSTORAGE_AWS_GLACIER_BUCKET_NAME";
+
+    protected static final String AWS_BUCKET_PREFIX_ENV_VAR = "COLDSTORAGE_AWS_BUCKET_PREFIX";
+
+    protected static final String AWS_REGION_ENV_VAR = "COLDSTORAGE_AWS_REGION";
+
+    protected final AWSCredentialsProvider awsCredentialsProvider = new DefaultAWSCredentialsProviderChain();
+
+    protected AWSCredentials credentials;
+
+    protected String mainBucket;
+
+    protected String glacierBucket;
+
+    protected String region;
+
+    protected String bucketPrefix;
+
```

Comment:
Could you add a private default constructor to avoid creation the helper from outside: 


```
private ColdStorageS3TestHelper() {
  // Not Allowed
}
```

Reply:
already have a default constructor